src/Uno.UI/UI/Xaml/Controls/Popup/PopupRoot.cs

```diff
@@ -19,6 +19,20 @@ internal partial class PopupRoot : Panel
 	public PopupRoot()
 	{
 		KeyDown += OnKeyDown;
+		Window.Current.Activated += (_, _) => CloseFlyouts();
+		Window.Current.SizeChanged += (_, _) => CloseFlyouts();
+	}
+
+	private void CloseFlyouts()
+	{
+		foreach (var reference in Enumerable.Reverse(_openPopups))
```

Comment:
Why is `Reverse` needed?

Reply:
Because WinUI does it that way IIUC. Yes, the tests I added will fail if Reverse is removed.

Reply:
Then consider `for (int i = _openPopups.Length - 1; i >= 0; i--)` as it's more efficient.